src/caw/caw_function.h

```diff
@@ -56,19 +56,33 @@ class CawFunction {
   static CawFuncReply Profile(const Any& payload, KeyValueInterface& kv);
 
   // TODO: WRITE FUNCTION DEFINITION WHEN COMPLETE
-  static CawFuncReply Stream(const Any& payload, KeyValueInterface& kv);
+  static CawFuncReply Stream(const Any& payload,
+                             const std::unordered_map<std::string, 
+                              std::vector<std::function<void(const Any&)>> >&
+                                current_streamers_);
 
   // map names of functions to functions
   static std::unordered_map<
       std::string, std::function<CawFuncReply(const Any&, KeyValueInterface&)> >
       function_map_;
 
+  // map names of stream_functions to functions
+  // Seperate because stream_function declaration is different
+  static std::unordered_map<
+      std::string, std::function<CawFuncReply(const Any&,
+                                              const std::unordered_map<std::string, 
```

Comment:
If this map/data type is used a lot you might want to add a type def.  This could also be a TODO for at the end of the project just to clean it up.

Reply:
Yup! I'll add it as a TODO for now.